src/Component/RequestBuilder/FacetSet.php

```diff
@@ -117,19 +117,30 @@ public function addFacetField($request, $facet)
             'facet.field',
             $this->renderLocalParams(
                 $field,
-                ['key' => $facet->getKey(), 'ex' => $facet->getExcludes()]
+                ['key' => $facet->getKey()
+                  , 'ex' => $facet->getExcludes()
+                  , 'facet.limit' => $facet->getLimit()
+                  , 'facet.sort' => $facet->getSort()
+                  , 'facet.prefix' => $facet->getPrefix()
+                  , 'facet.contains' => $facet->getContains()
+                  , 'facet.contains.ignoreCase' => $facet->getContainsIgnoreCase()
+                  , 'facet.offset' => $facet->getOffset()
+                  , 'facet.mincount' => $facet->getMinCount()
+                  , 'facet.missing' => $facet->getMissing()
+                  , 'facet.method' => $facet->getMethod()
+                ]
             )
         );
 
-        $request->addParam("f.$field.facet.limit", $facet->getLimit());
-        $request->addParam("f.$field.facet.sort", $facet->getSort());
-        $request->addParam("f.$field.facet.prefix", $facet->getPrefix());
-        $request->addParam("f.$field.facet.contains", $facet->getContains());
-        $request->addParam("f.$field.facet.contains.ignoreCase", $facet->getContainsIgnoreCase());
-        $request->addParam("f.$field.facet.offset", $facet->getOffset());
-        $request->addParam("f.$field.facet.mincount", $facet->getMinCount());
-        $request->addParam("f.$field.facet.missing", $facet->getMissing());
-        $request->addParam("f.$field.facet.method", $facet->getMethod());
+        //$request->addParam("f.$field.facet.limit", $facet->getLimit());
```

Comment:
We have failing tests on travis:
```
1) Solarium\Tests\Component\RequestBuilder\FacetSetTest::testBuildWithContainsSettings
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'?facet.field={!key=f1}owner&f.owner.facet.contains=foo&f.owner.facet.contains.ignoreCase=true&facet=true&facet.contains=bar&facet.contains.ignoreCase=false'
+'?facet.field={!key=f1 facet.contains=foo facet.contains.ignoreCase=1}owner&facet=true&facet.contains=bar&facet.contains.ignoreCase=false'
/home/travis/build/solariumphp/solarium/tests/Component/RequestBuilder/FacetSetTest.php:378
```

At least the failing test needs to be adjusted. But is the new output correct from your perspective?
```
?facet.field={!key=f1 facet.contains=foo facet.contains.ignoreCase=1}owner&facet=true&facet.contains=bar&facet.contains.ignoreCase=false
```